core/blockchain.go

```diff
@@ -1360,10 +1361,14 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 		log.Crit("Failed to write block into disk", "err", err)
 	}
 	// Commit all cached state changes into underlying memory database.
-	root, err := state.Commit(bc.chainConfig.IsEIP158(block.Number()))
-	if err != nil {
-		return NonStatTy, err
+	root, stateChanges, commitErr := state.Commit(bc.chainConfig.IsEIP158(block.Number()))
+	log.Debug("Sending StateChangeEvent to the feed", "block number", block.Number(), "count", len(stateChanges))
+	bc.stateChangeEventFeed.Send(StateChangeEvent{block, stateChanges})
```

Comment:
Good call!

Reply:
I've been struggling to find a way to test this. When I'm creating test transactions, there doesn't seem to be a case when there are 0 StateChanges, because I think the miner will always have their account balance increase. So I'm not sure that a guard here will do anything. Does that make sense? I'd be happy to rethink this/add a new story too!

Reply:
That makes sense to me!